pennylane/optimize/qnspsa.py

```diff
@@ -309,8 +309,8 @@ def _get_next_params(self, args, gradient):
 
         # params_vec and grad_vec group multiple inputs into the same vector to solve the
         # linear equation
-        params_vec = np.concatenate(params).reshape(-1)
-        grad_vec = np.concatenate(gradient).reshape(-1)
+        params_vec = np.concatenate([param.reshape(-1) for param in params])
+        grad_vec = np.concatenate([grad.reshape(-1) for grad in gradient])
```

Comment:
How come these were changed?

Reply:
This is equivalent to the old code in the case of old return types. In the case of new return types, NumPy doesn't allow concatenation of scalars (e.g. `np.concatenate([np.array(0.1), np.array(0.2)])` doesn't work). This change patches that behaviour to make it identical to the old return types.